.github/workflows/cli.yml

```diff
@@ -42,52 +42,125 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
-  test:
+  e2e-ubuntu:
     runs-on: ubuntu-22.04
     strategy:
       fail-fast: false
       matrix:
         shard: [1, 2, 3]
     steps:
+      - name: 🏗️ Setup Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: 18
+
+      - name: 🏗️ Setup Bun
+        uses: oven-sh/setup-bun@v2
+        with:
+          # Version `1.x` fails due to https://github.com/oven-sh/setup-bun/issues/37
+          # TODO(cedric): swap `latest` back once the issue is resolved
+          bun-version: latest
+
       - name: 👀 Checkout
         uses: actions/checkout@v4
         with:
           fetch-depth: 100
-      - uses: actions/setup-node@v3
-        with:
-          node-version: 18
       - name: ⬇️ Fetch commits from base branch
         run: git fetch origin ${{ github.event.before || github.base_ref || 'main' }}:${{ github.event.before || github.base_ref || 'main' }} --depth 100
         if: github.event_name == 'pull_request' || github.event_name == 'workflow_dispatch'
 
+      - name: 🔎 Find Yarn Cache
+        id: yarn-cache
+        run: echo "dir=$(yarn cache dir)" >> $GITHUB_OUTPUT
+        shell: bash
       - name: ♻️ Restore Yarn Cache
         uses: actions/cache@v4
         with:
-          path: ~/.cache/yarn
+          path: ${{ steps.yarn-cache.outputs.dir }}
           key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
           restore-keys: |
             ${{ runner.os }}-yarn-
 
       - name: 🧶 Install node modules in root dir
-        run: YARN_IGNORE_SCRIPTS=true yarn install --prefer-offline --frozen-lockfile
+        run: yarn install --prefer-offline --frozen-lockfile
+        env:
+          YARN_IGNORE_SCRIPTS: 'true'
       
-      - name: Build Expo CLI
+      - name: 👷 Build Expo CLI
         run: yarn workspace @expo/cli prepare
 
+      - name: 🔎 Type Check CLI
+        working-directory: packages/@expo/cli
+        run: yarn typecheck
+
+      - name: 🧪 E2E Test CLI
+        working-directory: packages/@expo/cli
+        run: yarn test:e2e --shard ${{ matrix.shard }}/${{ strategy.job-total }}
+
+  e2e-windows:
+    runs-on: windows-2022
+    continue-on-error: true
```

Comment:
TODO
- Add sharding once the E2E tests are passing on Windows, to speed things up without unnecessary spamming with error messages.
- Remove the `continue-on-error: true` once the test are passing on Windows